README.md

# Music Player Equalizer

Flutter_Equalizer_Custom is a flutter package which allow you to make your custom music player Equalizer.You can use this package to create a beautiful Equalizer without writing a bunch of code.






![App Screenshot](https://user-images.githubusercontent.com/112815459/229370288-02281b68-8570-406f-8cc0-7619116298ec.jpg)


## Documentation

[Documentation](https://github.com/Technicalranjitofficial/equalizer_flutter_custom)


## Installation

##### Add to pubspeck.yaml
    dependencies:
       equalizer_flutter_custom: ^0.0.3

    or

    flutter pub add equalizer_flutter_custom
 
## Quick synopsis 

```dart

CustomEqualizer(
     isEqEnabled: true, //Enable Equalizer
     playerSessionId:androidAudioSessionId!,//IMPORTANT AudioSessionId ),

```

#### Setup: Edit AndroidManifest.xml

      <manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="com.example.example">

      <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
      <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
      <uses-permission android:name="android.permission.MODIFY_AUDIO_SETTINGS" />

      </manifest>


#### Add This Dependency to pubspec.yml
    dependencies:
      provider: any



##### Wrap Your MaterialApp with ChangeNotifierProvider

```dart


@override
  Widget build(BuildContext context) {
    return ChangeNotifierProvider(
      create: (context) => EqualizerProvider(),// Import this from Current Package
      child: MaterialApp(
        title: 'Flutter Demo',
        debugShowCheckedModeBanner: false,
        theme: ThemeData(
          primarySwatch: Colors.blue,
        ),
        home: const MyHomePage(),
      ),
    );
  }

```


### Example with Complete Code

```dart

import 'package:flutter/material.dart';
import 'package:flutter_audio_query/flutter_audio_query.dart';
import 'package:just_audio/just_audio.dart';
import 'package:equalizer_flutter_custom/src/CustomEqualizer.dart';
import 'package:provider/provider.dart';
import 'package:equalizer_flutter_custom/src/EqalizerProvider/EqualizerProvider.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return ChangeNotifierProvider(
      create: (context) => EqualizerProvider(),
      child: MaterialApp(
        title: 'Demo Equalizer',
        debugShowCheckedModeBanner: false,
        theme: ThemeData(
          primarySwatch: Colors.blue,
        ),
        home: const MyHomePage(),
      ),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key});

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  late List<SongInfo> songs; //flutter_audio_query 

  late AudioPlayer audioPlayer; // using just_audio package. You can use your own.

  bool isLoading = true;
  getSOngs() async {
    FlutterAudioQuery audioQuery = FlutterAudioQuery();
    List<SongInfo> songsd = await audioQuery.getSongs();
    setState(() {
      songs = songsd;
      isLoading = false;
    });
  }

  @override
  void initState() {
    // TODO: implement initState
    audioPlayer = AudioPlayer();
    getSOngs();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text("Equalizer"),
      ),
      // body:
      body: isLoading
          ? const CircularProgressIndicator()
          : ListView.builder(
              itemCount: songs.length,
              itemBuilder: (context, index) {
                return InkWell(
                  onTap: () {
                    audioPlayer.setUrl(songs[index].filePath);
                    audioPlayer.play();
                    Future.delayed(const Duration(milliseconds: 300)) // Make Sure Your Audio must be playing before accessing the equalizer for the first time. You can handle your own function.
                        .then((value) {
                      Navigator.push(
                          context,
                          MaterialPageRoute(
                            builder: (context) =>  

                            //Equalizer_flutter_custom

                            CustomEqualizer( 
                              isEqEnabled: true, //Enable Equalizer
                              playerSessionId:
                                  audioPlayer.androidAudioSessionId!,// Most Important AudioSessionId NOT NULL
                              bandTextColor: Colors.green,
                              sliderBoxHeight: 220,
                              sliderBoxPadding: 10,
                              sliderBoxBorderRadius: BorderRadius.circular(20),
                            ),
                          ));
                    });
                  },
                  child: ListTile(
                    title: Text(songs[index].displayName),
                  ),
                );
              },
            ),
    );
  }
}

```


     

## Features

  - double? sliderBoxHeight; // Set the slider Box Height
  - Color? sliderBoxBackgroundColor; // Set SliderBox Bg Color
  - BorderRadius? sliderBoxBorderRadius; // Give borderRadius to Sliderbox
  - Color? inactiveTrackBar; // Change Slider inactiveTrackBar Color
  - Color? midPointBgColor;  // Change mid item Color
  - Color? midPointColorActive; // Mid item active color
  - Color? midPointColorInactive; // Mid item inActive color
  - IconData? midPointIcon; // set Custom Mid item Icons
  - Color? activeDisabledTrackBarColor; // set activeDisabledTrackBarColor 
  - Color? inactiveDisabledTrackBarColor; // set inactiveDisabledTrackBarColor color
  - Color? activeTrackBarMaxColor; // display custom color for high band
  - Color? activeTrackBarMinColor; // set active trackbarMinColor
  - Color? dragIndicatorActiveColor; // set drag handle active color
  - Color? dragIndicatorInactiveColor; //set drag handle inactive color
  - Color? bandTextColor; // custom text color for band
  - double? bandTextFontSize; // set custom text font for band text
  - TextStyle? presetsStyle; // band presets style
  - BoxDecoration? presetsBoxDecoration; // band box BoxDecoration
  - Color? baseTextColor; // bass text color
  - Widget? customSwitch; // Implement custom Switch instead of Default
  - bool isEqEnabled; // onn/off Equalizer
  - Color? appBarShadowColor; //appBarBgColor
  - String? appBarTitle; //appBarTitle custom
  - IconThemeData? actionsIconTheme;
  - bool? automaticallyImplyLeading;
  - Color? appBarBgColor;
  - double? appbarElevation;
  - Widget? appBarLeading;
  - TextStyle? titleTextStyle;
  - Color? inActiveSwitchTrackColor;
  - Color? activeSwitchTrackColor;
  - Color? activeSwitchColor;
  - Color? bgColor;- double? sliderBoxPadding;
## Color Reference

| Color             | Hex                                                                |
| ----------------- | ------------------------------------------------------------------ |
| Example Color | ![#0a192f](https://via.placeholder.com/10/0a192f?text=+) #0a192f |
| Example Color | ![#f8f8f8](https://via.placeholder.com/10/f8f8f8?text=+) #f8f8f8 |
| Example Color | ![#00b48a](https://via.placeholder.com/10/00b48a?text=+) #00b48a |
| Example Color | ![#00d1a0](https://via.placeholder.com/10/00b48a?text=+) #00d1a0 |


## License

MIT License

Copyright (c) 2023 Technicalranjit

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions

The above copyright notice and this permission notice shall be included in
all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
THE SOFTWARE.



CHANGELOG.md
## 0.0.3

# Issue related to slider-finxed


lib/src/Customeq.dart
import 'package:equalizer_flutter/equalizer_flutter.dart';
import 'package:equalizer_flutter_custom/src/CustomSlider.dart';
import 'package:flutter/material.dart';

Widget Customeq(
    min,
    max,
    double? sliderBoxHeight,
    Color? sliderBoxBackgroundColor,
    BorderRadiusGeometry? sliderBoxBorderRadius,
    Color? inactiveTrackBar,
    Color? midPointBgColor,
    Color? midPointColorActive,
    Color? midPointColorInactive,
    IconData? midPointIcon,
    Color? activeDisabledTrackBarColor,
    Color? inactiveDisabledTrackBarColor,
    Color? activeTrackBarMaxColor,
    Color? activeTrackBarMinColor,
    Color? dragIndicatorActiveColor,
    Color? dragIndicatorInactiveColor,
    Color? bandTextColor,
    double? bandTextFontSize,
    double? sliderBoxPadding) {
  int bandId = 0;
  return FutureBuilder<List<int>>(
    future: EqualizerFlutter.getCenterBandFreqs(),
    builder: (context, snapshot) {
      if (snapshot.connectionState == ConnectionState.done) {
        if (snapshot.hasData) {
          return Column(
            children: [
              Row(
                mainAxisAlignment: MainAxisAlignment.start,
                crossAxisAlignment: CrossAxisAlignment.start,
                children: snapshot.data!
                    .map((freq) => CustomSlider(
                          context,
                          bandId++,
                          freq,
                          min,
                          max,
                          sliderBoxHeight,
                          sliderBoxBackgroundColor,
                          sliderBoxBorderRadius,
                          inactiveTrackBar,
                          midPointBgColor,
                          midPointColorActive,
                          midPointColorInactive,
                          midPointIcon,
                          activeDisabledTrackBarColor,
                          inactiveDisabledTrackBarColor,
                          activeTrackBarMaxColor,
                          activeTrackBarMinColor,
                          dragIndicatorActiveColor,
                          dragIndicatorInactiveColor,
                          bandTextColor,
                          bandTextFontSize,
                          sliderBoxPadding,
                        ))
                    // .map((freq) => Container())
                    .toList(),
              )
            ],
          );
        } else {
          return const Text("Audio SessionId empty");
        }
      } else {
        return const CircularProgressIndicator();
      }
    },
  );
}


lib/src/CustomSlider.dart
import 'package:another_xlider/another_xlider.dart';
import 'package:another_xlider/models/handler.dart';
import 'package:another_xlider/models/trackbar.dart';
import 'package:equalizer_flutter/equalizer_flutter.dart';
import 'package:equalizer_flutter_custom/src/EqalizerProvider/EqualizerProvider.dart';
import 'package:equalizer_flutter_custom/src/Themes/GlobalThemes.dart';
import 'package:flutter/material.dart';

import 'package:provider/provider.dart';
// import 'package:sleek_circular_slider/sleek_circular_slider.dart';

GlobalThemes globalThemes = GlobalThemes();

Widget CustomSlider(
  context,
  int bandId,
  int freq,
  min,
  max,
  double? sliderBoxHeight,
  Color? sliderBoxBackgroundColor,
  BorderRadiusGeometry? sliderBoxBorderRadius,
  Color? inactiveTrackBar,
  Color? midPointBgColor,
  Color? midPointColorActive,
  Color? midPointColorInactive,
  IconData? midPointIcon,
  Color? activeDisabledTrackBarColor,
  Color? inactiveDisabledTrackBarColor,
  Color? activeTrackBarMaxColor,
  Color? activeTrackBarMinColor,
  Color? dragIndicatorActiveColor,
  Color? dragIndicatorInactiveColor,
  Color? bandTextColor,
  double? bandTextFontSize,
  double? sliderBoxPadding,
) {
  return Expanded(
    child: Padding(
      padding: sliderBoxPadding != null
          ? EdgeInsets.all(sliderBoxPadding)
          : const EdgeInsets.all(10),
      child: Column(
        children: [
          Column(
            children: [
              Container(
                height: sliderBoxHeight ?? 220,
                decoration: BoxDecoration(
                    color: sliderBoxBackgroundColor ?? Colors.grey.shade800,
                    // boxShadow: const [
                    //   // BoxShadow(
                    //   //   offset: const Offset(-2, -2),
                    //   //   color: globalThemes.colors2['boxShadowWhite']!,
                    //   // ),
                    //   // BoxShadow(
                    //   //   offset: const Offset(2, 2),
                    //   //   color: globalThemes.colors2['boxShadowDark']!,
                    //   // ),
                    // ],
                    borderRadius:
                        sliderBoxBorderRadius ?? BorderRadius.circular(20)),
                child: FutureBuilder<int>(
                  future: EqualizerFlutter.getBandLevel(bandId),
                  builder: (context, snapshot) {
                    if (snapshot.connectionState == ConnectionState.done) {
                      context
                          .read<EqualizerProvider>()
                          .setBandRange(bandId, snapshot.data!);
                      // print("BandInfo:bandId-${bandId}: ${snapshot.data}");
                      return Consumer<EqualizerProvider>(
                        builder: (context, value, child) {
                          return FlutterSlider(
                            trackBar: FlutterSliderTrackBar(
                              inactiveTrackBar: BoxDecoration(
                                  color:
                                      inactiveTrackBar ?? Colors.grey.shade700),
                              centralWidget: Container(
                                color: midPointBgColor ?? Colors.grey.shade800,
                                child: Icon(
                                  midPointIcon ?? Icons.drag_handle,
                                  color: value.isEqEnabled
                                      ? midPointColorActive ??
                                          Colors.cyan.shade700
                                      : midPointColorInactive ??
                                          Colors.grey.shade600,
                                ),
                              ),
                              activeDisabledTrackBarColor:
                                  activeDisabledTrackBarColor ??
                                      globalThemes.colors2['bgColor']!,
                              inactiveDisabledTrackBarColor:
                                  inactiveDisabledTrackBarColor ??
                                      globalThemes.colors2['bgColor']!,
                              activeTrackBar: BoxDecoration(
                                  color: value.bands[bandId].toDouble() >
                                          max / 2 - 3
                                      ? activeTrackBarMaxColor ??
                                          Colors.pink.shade600
                                      : activeTrackBarMinColor ?? Colors.cyan),
                            ),
                            axis: Axis.vertical,
                            jump: true,
                            visibleTouchArea: true,
                            // lockHandlers: true,
                            disabled: !value.isEqEnabled,

                            handler: FlutterSliderHandler(
                                child: Stack(
                              children: [
                                Container(
                                  // color: Colors.cyan,

                                  decoration: BoxDecoration(
                                      color: value.isEqEnabled
                                          ? Colors.cyan
                                          : Colors.grey.shade700,
                                      shape: BoxShape.rectangle,
                                      gradient: value.isEqEnabled
                                          ? globalThemes.LinearGradient1
                                          : globalThemes.LinearGradient2),

                                  // borderRadius:
                                  //     BorderRadius.all(Radius.circular(20))),
                                  // child: Icon(
                                  //   Icons.drag_indicator,
                                  //   color: value.isEqEnabled
                                  //       ? Colors.cyan
                                  //       : Colors.grey.shade600,
                                  // ),
                                ),
                                Container(
                                  height: 100,
                                  width: 100,
                                  color: Colors.black45,
                                  child: Icon(
                                    Icons.drag_indicator,
                                    size: 25,
                                    color: value.isEqEnabled
                                        ? dragIndicatorActiveColor ??
                                            Colors.cyan
                                        : dragIndicatorInactiveColor ??
                                            Colors.grey.shade600,
                                  ),
                                )
                              ],
                            )),
                            values: [value.bands[bandId].toDouble()],
                            rtl: true,
                            // visibleTouchArea: true,
                            max: max,
                            min: min,

                            // onDragging:
                            //     (handlerIndex, lowerValue, upperValue) async {

                            // },
                            onDragCompleted:
                                (handlerIndex, lowerValue, upperValue) {
                              value.changeFreq(bandId, lowerValue.toInt());
                            },
                          );
                        },
                      );
                    } else {
                      return const CircularProgressIndicator();
                    }
                  },
                ),
              ),
            ],
          ),
          Padding(
            padding: const EdgeInsets.only(top: 20),
            child: Text(
              "${(freq / 1000).toStringAsFixed(0)}HZ",
              style: TextStyle(
                  fontSize: bandTextFontSize ?? 12,
                  color: bandTextColor ?? globalThemes.colors2['textColor2']),
            ),
          )
        ],
      ),
    ),
  );
}


lib/src/CustomEqualizer.dart
import 'package:equalizer_flutter/equalizer_flutter.dart';
import 'package:equalizer_flutter_custom/src/EqalizerProvider/EqualizerProvider.dart';
import 'package:equalizer_flutter_custom/src/CustomSlider.dart';
import 'package:equalizer_flutter_custom/src/Customeq.dart';
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import 'package:sleek_circular_slider/sleek_circular_slider.dart';

class CustomEqualizer extends StatefulWidget {
  const CustomEqualizer(
      {super.key,
      required this.playerSessionId,
      this.sliderBoxHeight,
      this.sliderBoxBackgroundColor,
      this.sliderBoxBorderRadius,
      this.inactiveTrackBar,
      this.midPointBgColor,
      this.midPointColorActive,
      this.midPointColorInactive,
      this.midPointIcon,
      this.activeDisabledTrackBarColor,
      this.inactiveDisabledTrackBarColor,
      this.activeTrackBarMaxColor,
      this.activeTrackBarMinColor,
      this.dragIndicatorActiveColor,
      this.dragIndicatorInactiveColor,
      this.bandTextColor,
      this.bandTextFontSize,
      this.presetsStyle,
      this.presetsBoxDecoration,
      this.baseTextColor,
      this.customSwitch,
      required this.isEqEnabled,
      this.actionsIconTheme,
      this.appBarBgColor,
      this.appBarLeading,
      this.appBarShadowColor,
      this.appBarTitle,
      this.appbarElevation,
      this.automaticallyImplyLeading,
      this.titleTextStyle,
      this.inActiveSwitchTrackColor,
      this.activeSwitchTrackColor,
      this.activeSwitchColor,
      this.bgColor,
      this.sliderBoxPadding});

  final int playerSessionId;

  final double? sliderBoxHeight;
  final Color? sliderBoxBackgroundColor;
  final BorderRadius? sliderBoxBorderRadius;
  final Color? inactiveTrackBar;
  final Color? midPointBgColor;
  final Color? midPointColorActive;
  final Color? midPointColorInactive;
  final IconData? midPointIcon;
  final Color? activeDisabledTrackBarColor;
  final Color? inactiveDisabledTrackBarColor;
  final Color? activeTrackBarMaxColor;
  final Color? activeTrackBarMinColor;
  final Color? dragIndicatorActiveColor;
  final Color? dragIndicatorInactiveColor;
  final Color? bandTextColor;
  final double? bandTextFontSize;
  final TextStyle? presetsStyle;
  final BoxDecoration? presetsBoxDecoration;
  final Color? baseTextColor;
  final Widget? customSwitch;
  final bool isEqEnabled;
  final Color? appBarShadowColor;
  final String? appBarTitle;
  final IconThemeData? actionsIconTheme;
  final bool? automaticallyImplyLeading;
  final Color? appBarBgColor;
  final double? appbarElevation;
  final Widget? appBarLeading;
  final TextStyle? titleTextStyle;
  final Color? inActiveSwitchTrackColor;
  final Color? activeSwitchTrackColor;
  final Color? activeSwitchColor;
  final Color? bgColor;

  final double? sliderBoxPadding;

  @override
  State<CustomEqualizer> createState() => _CustomEqualizerState();
}

class _CustomEqualizerState extends State<CustomEqualizer> {
  @override
  void initState() {
    // TODO: implement initState
    iniaudio();
    super.initState();
    // EqualizerFlutter.setAudioSessionId(widget.PlayerSessionId);

    WidgetsBinding.instance.addPostFrameCallback((_) => deffun());
  }

  iniaudio() async {
    await EqualizerFlutter.init(widget.playerSessionId);
    context.read<EqualizerProvider>().setEqEnabled(widget.isEqEnabled);
    await context.read<EqualizerProvider>().getPresest();
  }

  deffun() async {
    // if (context.read<EqualizerProvider>().isEqEnabled) {

    context.read<EqualizerProvider>().setEqEnabled(widget.isEqEnabled);
    // }
  }

  String dropdownValue = 'Dog';

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        appBar: AppBar(
          title: Text(widget.appBarTitle ?? "Equalizer"),
          shadowColor: widget.appBarShadowColor,
          actionsIconTheme: widget.actionsIconTheme,
          automaticallyImplyLeading: widget.automaticallyImplyLeading ?? true,
          backgroundColor:
              widget.appBarBgColor ?? globalThemes.colors2['bgColor'],
          elevation: widget.appbarElevation,
          leading: widget.appBarLeading,
          titleTextStyle: widget.titleTextStyle,
          actions: [
            Consumer<EqualizerProvider>(
              builder: (context, value, child) {
                return widget.customSwitch ??
                    Switch(
                      value: value.isEqEnabled,
                      onChanged: (data) {
                        value.isEqEnabled
                            ? value.setEqEnabled(false)
                            : value.setEqEnabled(true);
                      },
                      inactiveTrackColor: widget.inActiveSwitchTrackColor ??
                          Colors.grey.shade600,
                      activeColor: widget.activeSwitchColor ?? Colors.teal,
                      activeTrackColor:
                          widget.activeSwitchTrackColor ?? Colors.green,
                    );
              },
            )
          ],
        ),
        body: Container(
          decoration: BoxDecoration(
              color: widget.bgColor ?? globalThemes.colors2['bgColor']),
          height: MediaQuery.of(context).size.height,
          child: Column(
            // mainAxisAlignment: MainAxisAlignment.spaceBetween,
            children: [
              const SizedBox(
                height: 10,
              ),
              SizedBox(
                  width: MediaQuery.of(context).size.width,
                  child: FutureBuilder(
                    future: EqualizerFlutter.getBandLevelRange(),
                    builder: (context, snapshot) {
                      if (snapshot.connectionState == ConnectionState.done) {
                        if (snapshot.hasData) {
                          return Customeq(
                              snapshot.data![0].toDouble(),
                              snapshot.data![1].toDouble(),
                              widget.sliderBoxHeight,
                              widget.sliderBoxBackgroundColor,
                              widget.sliderBoxBorderRadius,
                              widget.inactiveTrackBar,
                              widget.midPointBgColor,
                              widget.midPointColorActive,
                              widget.midPointColorInactive,
                              widget.midPointIcon,
                              widget.activeDisabledTrackBarColor,
                              widget.inactiveDisabledTrackBarColor,
                              widget.activeTrackBarMaxColor,
                              widget.activeTrackBarMinColor,
                              widget.dragIndicatorActiveColor,
                              widget.dragIndicatorInactiveColor,
                              widget.bandTextColor,
                              widget.bandTextFontSize,
                              widget.sliderBoxPadding);
                        } else {
                          return const Text("AudioSessinId Not Found");
                        }
                      } else {
                        return const CircularProgressIndicator();
                      }
                    },
                  )),
              const Spacer(),
              SizedBox(
                width: MediaQuery.of(context).size.width - 100,
                child: Row(
                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
                  children: [
                    Column(
                      children: [
                        Consumer<EqualizerProvider>(
                          builder: (context, value, child) {
                            return value.isEqEnabled
                                ? SleekCircularSlider(
                                    min: 0,
                                    max: 1000,
                                    initialValue: value.bassval,
                                    onChangeEnd: (val) {
                                      value.setbassval(val);
                                    },
                                  )
                                : SleekCircularSlider(
                                    min: 0,
                                    max: 1000,
                                    initialValue: value.bassval,
                                    appearance: CircularSliderAppearance(
                                        customColors: CustomSliderColors(
                                      dotColor: Colors.grey.shade500,
                                      dynamicGradient: false,
                                      progressBarColor: Colors.grey.shade500,
                                      trackColor: Colors.grey.shade500,
                                    )),
                                  );
                          },
                        ),
                        Text(
                          "Bass",
                          style: TextStyle(
                              color: widget.baseTextColor ??
                                  globalThemes.colors2['textColor2']),
                        )
                      ],
                    ),
                    Consumer<EqualizerProvider>(
                      builder: (context, value, child) {
                        print(value.currentp);
                        return FutureBuilder(
                            future: value.getPresest(),
                            builder: (context, snapshot) {
                              if (snapshot.hasData) {
                                return Container(
                                  decoration: widget.presetsBoxDecoration,
                                  width: 60,
                                  child: IgnorePointer(
                                    ignoring: !value.isEqEnabled,
                                    child: DropdownButton<String>(
                                      // Step 3
                                      // .

                                      alignment: Alignment.center,
                                      isExpanded: true,

                                      value: value.currentp,

                                      // Step 4.

                                      items: snapshot.data!
                                          .map<DropdownMenuItem<String>>(
                                              (String value) {
                                        return DropdownMenuItem<String>(
                                          value: value,
                                          child: Text(
                                            value,
                                            style: widget.presetsStyle ??
                                                const TextStyle(
                                                  fontSize: 10,
                                                  color: Colors.teal,
                                                ),
                                          ),
                                        );
                                      }).toList(),
                                      onChanged: (v) {
                                        value.setPresets(v);
                                        setState(() {});
                                      },
                                    ),
                                  ),
                                );
                              } else {
                                return const CircularProgressIndicator();
                              }
                            });

                        // return Text(
                        //   value.custom ? "Custom" : value.CurrentPresets,
                        //   style: TextStyle(
                        //       color: globalThemes.colors2['textColor2']),
                        // );
                      },
                    )
                  ],
                ),
              ),
              const Spacer()
            ],
          ),
        ));
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:flutter_audio_query/flutter_audio_query.dart';
import 'package:just_audio/just_audio.dart';
import 'package:equalizer_flutter_custom/src/CustomEqualizer.dart';
import 'package:provider/provider.dart';
import 'package:equalizer_flutter_custom/src/EqalizerProvider/EqualizerProvider.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return ChangeNotifierProvider(
      create: (context) => EqualizerProvider(),
      child: MaterialApp(
        title: 'Demo Equalizer',
        debugShowCheckedModeBanner: false,
        theme: ThemeData(
          primarySwatch: Colors.blue,
        ),
        home: const MyHomePage(),
      ),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key});

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  late List<SongInfo> songs;

  late AudioPlayer audioPlayer;

  bool isLoading = true;
  getSOngs() async {
    FlutterAudioQuery audioQuery = FlutterAudioQuery();
    List<SongInfo> songsd = await audioQuery.getSongs();
    setState(() {
      songs = songsd;
      isLoading = false;
    });
  }

  @override
  void initState() {
    // TODO: implement initState
    audioPlayer = AudioPlayer();
    getSOngs();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text("Equalizer"),
      ),
      // body:
      body: isLoading
          ? const CircularProgressIndicator()
          : ListView.builder(
              itemCount: songs.length,
              itemBuilder: (context, index) {
                return InkWell(
                  onTap: () {
                    audioPlayer.setUrl(songs[index].filePath);
                    audioPlayer.play();
                    Future.delayed(const Duration(milliseconds: 300))
                        .then((value) {
                      Navigator.push(
                          context,
                          MaterialPageRoute(
                            builder: (context) => CustomEqualizer(
                              isEqEnabled: true,
                              playerSessionId:
                                  audioPlayer.androidAudioSessionId!,
                              bandTextColor: Colors.green,
                              sliderBoxHeight: 220,
                              sliderBoxPadding: 10,
                              appbarElevation: 0,
                              appBarShadowColor: Colors.grey,
                              titleTextStyle:
                                  const TextStyle(color: Colors.red),
                              sliderBoxBorderRadius: BorderRadius.circular(20),
                            ),
                          ));
                    });
                  },
                  child: ListTile(
                    title: Text(songs[index].displayName),
                  ),
                );
              },
            ),
    );
  }
}
